src/wesnoth.cpp

```diff
@@ -73,16 +73,13 @@
 
 #include <boost/iostreams/categories.hpp>   // for input, output
 #include <boost/iostreams/copy.hpp>         // for copy
-#include <boost/iostreams/filter/bzip2.hpp> // for bzip2_compressor, etc
 
 #if defined(_MSC_VER)
 #pragma warning(push)
 #pragma warning(disable : 4456)
 #pragma warning(disable : 4458)
 #endif
 
-#include <boost/iostreams/filter/gzip.hpp> // for gzip_compressor, etc
-
 #if defined(_MSC_VER)
 #pragma warning(pop)
```

Comment:
This one's learnt from experience, but these `#ifdef _MSC_VER #pragma ...` lines are turning two specific warnings off because they're triggered by the gzip.hpp file, and then the `pop` is turning them back on again.

All of these should be removed along with the `#include <.../gzip.hpp>` line.